packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -514,6 +548,35 @@ class Scaffold extends StatefulWidget {
     );
   }
 
+  /// Returns a [ValueNotifier] for the [ScaffoldGeometry] for the closest
+  /// [Scaffold] ancestor of the given context.
+  ///
+  /// Value notifications are sent during the layout pass.
+  static ValueNotifier<ScaffoldGeometry> geometryOf(BuildContext context) {
```

Comment:
to avoid anyone being tempted to set the value directly on this object, maybe make this return a ValueListenable instead of a ValueNotifier. (ValueNotifier implements ValueListenable so no code need change.)